kafka/src/main/java/io/cloudevents/kafka/impl/KafkaBinaryMessageReaderImpl.java

```diff
@@ -44,7 +44,7 @@ protected boolean isContentTypeHeader(String key) {
 
     @Override
     protected boolean isCloudEventsHeader(String key) {
-        return key.length() > 3 && key.substring(0, KafkaHeaders.CE_PREFIX.length()).startsWith(KafkaHeaders.CE_PREFIX);
+        return key.length() > KafkaHeaders.CE_PREFIX.length() && key.startsWith(KafkaHeaders.CE_PREFIX);
```

Comment:
Any particular reason this isn't using your new method ?

Also ... Might be worth adding the null-check on `key` (most of the other readers do this)

Reply:
> Any particular reason this isn't using your new method ?

it wasn't case-insensitive before like other ones and I think the reason is that other methods are HTTP based (headers normalization) vs this one being Kafka based. 

Do you know if the spec is case-insensitive for every defined protocol binding?

Honestly, from a user perspective, I would be surprised if a header isn't considered a CE header due to the case, so I agree with you.

Reply:
Hey @JemDay since most of the comments aren't strictly related to the original goal of the PR and the code is strictly equivalent to the existing one, I think, we can follow up on these comments in a different PR/issue.